File: ios/Models/ThermalEvent.swift
import Foundation

struct ThermalEvent: Codable {
    enum Severity: String, Codable {
        case info
        case warning
        case critical
    }

    enum ActionTaken: String, Codable {
        case logOnly
        case promptUser
        case autoReduceFeatures
    }

    let timestamp: Date
    let severity: Severity
    let actionTaken: ActionTaken
}

File: android/app/src/main/java/com/golfiq/hud/model/ThermalEvent.kt
package com.golfiq.hud.model

import java.time.Instant

data class ThermalEvent(
    val timestamp: Instant,
    val severity: Severity,
    val actionTaken: ActionTaken,
) {
    enum class Severity { INFO, WARNING, CRITICAL }
    enum class ActionTaken { LOG_ONLY, PROMPT_USER, AUTO_REDUCE_FEATURES }
}